### ISSUE TITLE
 class NetworkPreparationProcedure

### ISSUE THREAD

**Issue by asxzswqaz** [AequilibraE/qaequilibrae#302]
Excuse me for my bad coding. But I believe in the NetworkPreparationProcedure class, the duplication_layer method should be in the class.

Cheers

**Comment by pedrocamargo** [comment]
Can you substantiate what you are saying? 

I will also ask you to only post on one of our communication channels at a time ( I have not accepted the question in the mailing list as it is not relevant due to its lack of context and/or substantiation)

**Comment by asxzswqaz** [comment]
Hey Pedro, 

Sorry for the bad post - I was in a rush yesterday, and didn't get a chance to post the question correctly. 

First of all, massive thanks for putting together AequilibraE and making it open source – it's a fantastic tool for all of us in the community.

I've been messing around with AequilibraE in QGIS, specifically trying out the network manipulation tool. Ran into a bit of a hiccup, though. When I was setting up my network using the network preparation tool, the progress bar wouldn't move from 0%, and this error popped up in the log about a missing duplication_layer method in the NetworkPreparationProcedure class.

So, I got curious and poked around in the source code. Looks like there might be a tiny indentation error that's causing the duplication_layer method to sit outside the NetworkPreparationProcedure class instead of inside it (image attached).
![image](https://github.com/AequilibraE/qaequilibrae/assets/80319745/813f10b8-2e20-40d6-8055-6047e3581033)


I'm still pretty new to coding, so if I'm off base here, just let me know. But when I tweaked the indentation on my local drive, the error disappeared. Thought I'd pass this on to you in case it's a helpful fix for the project.

Thanks again for all the awesome work on AequilibraE!

Cheers,
